app/controllers/posts_controller.rb

```diff
@@ -5,94 +5,60 @@
 class PostsController < ApplicationController
   include PostsHelper
 
-  before_action :authenticate_user!, :except => [:show, :iframe, :oembed, :interactions]
-  before_action :set_format_if_malformed_from_status_net, :only => :show
-  before_action :find_post, :only => [:show, :interactions]
+  before_action :authenticate_user!, only: :destroy
+  before_action :set_format_if_malformed_from_status_net, only: :show
 
-  respond_to :html,
-             :mobile,
-             :json,
-             :xml
+  respond_to :html, :mobile, :json, :xml
 
-  rescue_from Diaspora::NonPublic do |exception|
+  rescue_from Diaspora::NonPublic do |_exception|
     respond_to do |format|
-      format.all { render :template=>'errors/not_public', :status=>404, :layout => "application"}
+      format.all { render template: "errors/not_public", status: 404, layout: "application" }
     end
   end
 
   def show
-    mark_corresponding_notifications_read if user_signed_in?
-
+    post_service = PostService.new(id: params[:id], user: current_user)
+    post_service.assign_post_and_mark_notifications
+    @post = post_service.post
     respond_to do |format|
-      format.html {
-        gon.post = PostPresenter.new(@post, current_user)
-        render "posts/show"
-      }
+      format.html { gon.post = post_service.present_json }
       format.xml { render xml: @post.to_diaspora_xml }
-      format.mobile { render "posts/show" }
-      format.json { render json: PostPresenter.new(@post, current_user) }
+      format.json { render json: post_service.present_json }
     end
   end
 
   def iframe
-    render :text => post_iframe_url(params[:id]), :layout => false
+    render text: post_iframe_url(params[:id]), layout: false
   end
 
   def oembed
     post_id = OEmbedPresenter.id_from_url(params.delete(:url))
-    post = Post.find_by_guid_or_id_with_user(post_id, current_user)
-    if post.present?
-      oembed = OEmbedPresenter.new(post, params.slice(:format, :maxheight, :minheight))
-      render :json => oembed
-    else
-      render :nothing => true, :status => 404
-    end
+    post_service = PostService.new(id: post_id, user: current_user,
+                                    oembed: params.slice(:format, :maxheight, :minheight))
+    post_service.assign_post
+    render json: post_service.present_oembed
   end
 
   def interactions
-    respond_with(PostInteractionPresenter.new(@post, current_user))
+    post_service = PostService.new(id: params[:id], user: current_user)
+    post_service.assign_post
+    respond_with(post_service.present_interactions_json)
```

Comment:
Both, my personal rule is to omit them almost everywhere possible, in shared codebases I leave them for method definitions at least.